### PR TITLE
Load data in event form show notifications

### PR DESCRIPTION
**Pull request by Andreslw**
Load data in event-form
Update event info
add notification when an event is created or updated,
add notification when fields are empty, before save.

### CODE REVIEW

File: src/partner/modules/ui/Modal/Notification.tsx [hunk 1]
```diff
@@ -1,42 +1,52 @@
...
+  line-height: 1.33;
 `;
 
 export const Notification: React.SFC<INotificationProps> = props => {
   return (
     <NotificacionDiv>
-      <div>{props.text}</div>
-      <ButtonClose onClick={() => props.close()}>OK</ButtonClose>
+      <TextNotification>{props.text}</TextNotification>
+      <ButtonClose onClick={() => props.close()}>
```
Comment (mishelashala) [inline src/partner/modules/ui/Modal/Notification.tsx]:
```suggestion
      <ButtonClose onClick={props.close}>
```

File: src/partner/services/EventService.ts [hunk 2]
```diff
@@ -49,6 +52,20 @@ export const eventService: IEventService = {
...
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/v1/events/${eventId}`, getConfig);
+      const data = await res.json();
+      if (data.statusCode === 200) {
+        return EventMapper.toEntity(data.data);
+      } else {
+        throw new Error("Error at getting the event");
+      }
+    } catch (err) {
+      console.error(err);
```
Comment (mishelashala) [inline src/partner/services/EventService.ts]: delete logs

File: src/partner/services/EventService.ts [hunk 3]
```diff
@@ -65,4 +82,21 @@ export const eventService: IEventService = {
       console.error(err);
     }
   },
+  putEvent: async (event: IEvent) => {
+    try {
+      const eventId = event.id;
```
Comment (mishelashala) [inline src/partner/services/EventService.ts]:
```suggestion
      const { id as eventId, ...eventEntity } = event;
```

File: src/partner/services/EventService.ts [hunk 4]
```diff
@@ -65,4 +82,21 @@ export const eventService: IEventService = {
...
+      delete event["id"]; //not allowed by the server
```
Comment (mishelashala) [inline src/partner/services/EventService.ts]:
```suggestion
```

File: src/partner/services/EventService.ts [hunk 5]
```diff
@@ -65,4 +82,21 @@ export const eventService: IEventService = {
...
+
+      putConfig.body = JSON.stringify(EventMapper.toDTO(event));
```
Comment (mishelashala) [inline src/partner/services/EventService.ts]:
```suggestion
      putConfig.body = JSON.stringify(EventMapper.toDTO(eventEntity));
```

File: src/partner/services/EventService.ts [hunk 6]
```diff
@@ -65,4 +82,21 @@ export const eventService: IEventService = {
...
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/v1/events/${eventId}`, putConfig);
+      const data = await res.json();
+      if (data.statusCode === 201) {
```
Comment (mishelashala) [inline src/partner/services/EventService.ts]: use early return

```js
if (data.statusCode !== 200) {
  throw new Error("Error at updating event");
}

return data;
```

File: src/partner/views/CreateEvent/CreateEvent.tsx [hunk 7]
```diff
@@ -54,16 +73,48 @@ export const CreateEvent: React.FC<RouteComponentProps> = ({ history }) => {
...
+
   const handleSaveEvent = async () => {
-    const res = await eventService.postEvent({ ...state });
+    const errors = getFieldErrors();
+    if (errors.length > 0) {
+      notificationContext.handleShowNotification(errors.join(", "));
+      return;
+    }
+
+    let res;
```
Comment (mishelashala) [inline src/partner/views/CreateEvent/CreateEvent.tsx]: use ternary instead

```js
let res = isEditing ? await eventService.putEvent({ ...state }) : await eventService.postEvent({ ...state })
```

File: src/partner/views/CreateEvent/CreateEvent.tsx [hunk 8]
```diff
@@ -54,16 +73,48 @@ export const CreateEvent: React.FC<RouteComponentProps> = ({ history }) => {
...
+    }
+
+    return errors;
+  };
+
   const handleSaveEvent = async () => {
-    const res = await eventService.postEvent({ ...state });
+    const errors = getFieldErrors();
+    if (errors.length > 0) {
+      notificationContext.handleShowNotification(errors.join(", "));
```
Comment (mishelashala) [inline src/partner/views/CreateEvent/CreateEvent.tsx]: keep consistency:

```suggestion
      return notificationContext.handleShowNotification(errors.join(", "));
```

File: src/partner/views/CreateEvent/CreateEvent.tsx [hunk 9]
```diff
@@ -54,16 +73,48 @@ export const CreateEvent: React.FC<RouteComponentProps> = ({ history }) => {
...
+      return;
```
Comment (mishelashala) [inline src/partner/views/CreateEvent/CreateEvent.tsx]:
```suggestion
```

**Review by mishelashala — CHANGES_REQUESTED** [review]

File: src/user/services/OrderService.tsx [hunk 10]
```diff
@@ -12,7 +12,7 @@ export const orderService: IOrderService = {
       const data = await res.json();
       return data.orders.map(orderMapper.toModel);
     } catch (err) {
-      console.log(err);
+      console.error(err);
```
Comment (Andreslw) [inline src/user/services/OrderService.tsx:15]: omit this one

**Review by mishelashala — APPROVED** [review]

**Pull request merged**